dev-libs/libpipeline/libpipeline-1.5.0.recipe

```diff
@@ -0,0 +1,72 @@
+SUMMARY="A C library for setting up and running pipelines"
+DESCRIPTION="libpipeline is a C library for setting up and running pipelines \
+of processes, without needing to involve shell command-line parsing which is \
+often error-prone and insecure.  This alleviates programmers of the need to \
+laboriously construct pipelines using lower-level primitives such as fork(2) \
+and execve(2)."
+HOMEPAGE="http://libpipeline.nongnu.org/"
+COPYRIGHT="1994 Markus Armbruster.
+	1989, 1990, 1991, 1992, 2000, 2001, 2002, 2003, 2004, 2005,
+	2006, 2007, 2008, 2009, 2010 Free Software Foundation, Inc.
+	2003-2017 Colin Watson"
+LICENSE="GNU GPL v3"
+REVISION="1"
+SOURCE_URI="http://download.savannah.nongnu.org/releases/libpipeline/libpipeline-$portVersion.tar.gz"
+CHECKSUM_SHA256="0d72e12e4f2afff67fd7b9df0a24d7ba42b5a7c9211ac5b3dcccc5cd8b286f2b"
+
+ARCHITECTURES="x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	libpipeline$secondaryArchSuffix = $portVersion
+	lib:libpipeline$secondaryArchSuffix = $portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	libpipeline${secondaryArchSuffix}_devel = $portVersion
+	devel:libpipeline$secondaryArchSuffix = $portVersion
+	"
+REQUIRES_devel="
+	libpipeline$secondaryArchSuffix == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	"
+BUILD_PREREQUIRES="
+	cmd:aclocal
+	cmd:autoconf
+	cmd:awk
+	cmd:gcc$secondaryArchSuffix
+	cmd:libtoolize$secondaryArchSuffix
+	cmd:make
+	cmd:pkg_config$secondaryArchSuffix
+	"
+
+defineDebugInfoPackage libpipeline$secondaryArchSuffix \
+	$libDir/libpipeline.so.$portVersion
+
+BUILD()
+{
+	autoreconf -vfi
+	runConfigure ./configure
+	make $jobArgs
+}
+
+INSTALL()
+{
+	make install
+
+	#remove libtool file
+	rm $libDir/*.la
+
+	prepareInstalledDevelLib libpipeline
+	fixPkgconfig
+
+	# devel package
+	packageEntries devel \
+		$developDir
```

Comment:
I've just tried building libpipeline and noticed that the man3 pages are in the main package.
It would be nice to move $manDir/man3 to the devel package.
You can do so by replacing
```
	# devel package
	packageEntries devel \
		$developDir
```
by
```
	if [ -n "$secondaryArchSuffix" ]; then
		rm -rf $documentationDir
		maybe_manDir_man3=
	else
		maybe_manDir_man3=$manDir/man3
	fi

	# devel package
	packageEntries devel \
		$developDir \
		$maybe_manDir_man3
```
Other benefit: it will get rid of the docs on secondary arch builds ;-)

Reply:
This doesn't work if you're building for $secondaryArchSuffix?

Reply:
Oops, you are perfectly right. My mistake!
This should work:
```
	if [ -n "$secondaryArchSuffix" ]; then
		rm -rf $manDir
		maybe_manDir=
	else
		maybe_manDir=$manDir
	fi

	# devel package
	packageEntries devel \
		$developDir \
		$maybe_manDir

	rmdir $documentationDir
```

Actually, when I noticed that the only man section was man3, I though we could simplify the INSTALL without having to use the maybe_manDir... variable, but I did not test the "simplified" version...

BTW, I did not test what I wrote above.

Reply:
One question here, why remove the $mandir for $secondaryArchSuffix?

Reply:
Sorry for this late reply. I guess you already found the answer. Here it is, for the other readers who might not yet know the answer.
We don't need the docs on secondary arch builds because the recipe builds fine on *all* primary architectures.